src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -257,8 +258,331 @@ private struct IntPtrAndHandle
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        private static unsafe IntPtr GetMarshalMethodPointer (Type type, string name, out Type? returnType, out Type? parameterType) {
+            var info = type.GetMethod(
+                name, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic
+            );
+            if (info == null) {
+                parameterType = null;
+                returnType = null;
+                return IntPtr.Zero;
+            }
+
+            var p = info.GetParameters();
+            if (p.Length != 1)
+                throw new Exception($"Method {type.Name}.{name} must accept exactly one parameter");
+
+            parameterType = p[0].ParameterType;
+            returnType = info.ReturnType;
+
+            var tmp = default(IntPtrAndHandle);
+            tmp.handle = info.MethodHandle;
+            return tmp.ptr;
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2057:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        public static unsafe string GetCustomMarshalerInfoForType (IntPtr typePtr, string marshalerFullName) {
+            if (typePtr == IntPtr.Zero)
+                return "null";
+            else if (string.IsNullOrEmpty(marshalerFullName))
+                return "null";
+
+            IntPtrAndHandle tmp = default(IntPtrAndHandle);
+            tmp.ptr = typePtr;
+            var typeHandle = tmp.typeHandle;
+
+            var type = Type.GetTypeFromHandle(typeHandle);
+            var marshalerType = Type.GetType(marshalerFullName) ?? type.Assembly.GetType(marshalerFullName);
+            if (marshalerType == null)
+                return "null";
+
+            var preFilter = GetAndEscapeStringProperty(marshalerType, "FromJavaScriptPreFilter");
+            var postFilter = GetAndEscapeStringProperty(marshalerType, "ToJavaScriptPostFilter");
+
+            var inputPtr = GetMarshalMethodPointer(marshalerType, "FromJavaScript", out Type? fromReturnType, out Type? fromParameterType);
```

Comment:
The parameter type can be only string, why is it extracted and never checked?

Reply:
It can be any JS type, not string